tensorflow_probability/python/bijectors/gev_cdf.py

```diff
@@ -0,0 +1,218 @@
+    * exp(-(x - loc) / scale) when conc = 0.
+  ```
+  """
+
+  def __init__(self,
+               loc=0.,
+               scale=1.,
+               concentration=0,
+               validate_args=False,
+               name='generalizedextremevalue_cdf'):
+    """Instantiates the `GeneralizedExtremeValueCDF` bijector.
+
+    Args:
+      loc: Float-like `Tensor` that is the same dtype and is
+        broadcastable with `scale` and `concentration`.
+      scale: Positive Float-like `Tensor` that is the same dtype and is
+        broadcastable with `loc` and `concentration`.
+      concentration: Nonzero float-like `Tensor` that is the same dtype and is
+        broadcastable with `loc` and `scale`.
+      validate_args: Python `bool` indicating whether arguments should be
+        checked for correctness.
+      name: Python `str` name given to ops managed by this object.
+    """
+    parameters = dict(locals())
+    with tf.name_scope(name) as name:
+      dtype = dtype_util.common_dtype(
+          [loc, scale, concentration], dtype_hint=tf.float32)
+      self._loc = tensor_util.convert_nonref_to_tensor(
+          loc, dtype=dtype, name='loc')
+      self._scale = tensor_util.convert_nonref_to_tensor(
+          scale, dtype=dtype, name='scale')
+      self._concentration = tensor_util.convert_nonref_to_tensor(
+          concentration, dtype=dtype, name='concentration')
+      super(GeneralizedExtremeValueCDF, self).__init__(
+          validate_args=validate_args,
+          forward_min_event_ndims=0,
+          parameters=parameters,
+          name=name)
+
+  @property
+  def loc(self):
+    """
+    The `loc` in `Y = exp(-t(X))`
+    where `t(x) = (1 + conc * (x - loc) / scale) ) ** (-1 / conc)`
+    when `conc != 0` and `exp(-(x - loc) / scale)` when `conc = 0`.`
+    """
+    return self._loc
+
+  @property
+  def scale(self):
+    """
+    The `scale` in `Y = exp(-t(X))`
+    where `t(x) = (1 + conc * (x - loc) / scale) ) ** (-1 / conc)`
+    when `conc != 0` and `exp(-(x - loc) / scale)` when `conc = 0`.`
+    """
+    return self._scale
+
+  @property
+  def concentration(self):
+    """
+    The `concentration` in `Y = exp(-t(X))`
+    where `t(x) = (1 + conc * (x - loc) / scale) ) ** (-1 / conc)`
+    when `conc != 0` and `exp(-(x - loc) / scale)` when `conc = 0`.`
+    """
+    return self._concentration
+
+  @classmethod
+  def _is_increasing(cls):
+    return True
+
+  def _forward(self, x):
+    with tf.control_dependencies(self._maybe_assert_valid_x(x)):
+      conc = tf.convert_to_tensor(self.concentration)
+
+      z = (x - self.loc) / self.scale
+
+      # for the numerical stability, we set a threshold `tol` that
+      # when `abs(conc) < tol`, gev converges to the Gumbel dist.
+      # check the following link for the estimation
+      # of the threshold (also for _inverse and _forward_log_det_jacobian)
+      # https://github.com/tensorflow/probability/pull/901#issuecomment-629857187
+      tol_target = 1e-5
```

Comment:
Note this tolerance needs to be dtype dependent (float64 vs float32). Do you think you can change that? I am also fine with dropping the tolerance stuff at the moment to get this checked in and later iterate on stability near 0.